You#21

KS-you opened the pull request:
#18 
マージしてしまった内容のrevertがうまくいかなかったので、修正した内容を上げ直します。
以下のモジュールをversionをつけて編集してます。

pythoh-dotenv==1.0.1
※ Pythonファイルに.envを読み込ませるモジュール

再度確認お願いします。

KS-you committed:
requirements.txtの修正

mo-ri-regen commented on back/stretch_app/__init__.py:
```diff
@@ -0,0 +1,50 @@
+        'mysql://{user}:{password}@{host}/{db_name}?charset=utf8'.format(**{
+        'user': "stretch_user",
+        'password': "stretch_MySQL",
+        'host': 'localhost',
```
31行目　パスワードバレているけど問題ない？

mo-ri-regen commented on back/stretch_app/setup.py:
```diff
@@ -0,0 +1,6 @@
+flask_stretch_app = create_app()
+
+if __name__ == '__main__':
+    flask_stretch_app.run(debug=True, host='0.0.0.0', port=5001)
```
6行目は変更されていないけど、本番環境にデプロイしたら通らないから
気にしなくていいってことですかね？

mo-ri-regen commented:
@KS-you 
envファイルのサンプルファイル
(たとえば.env.local(ファイル名はローカルのやつってわかればいいのであまり決まりはないです)のようなもの)もpushお願いします🙏

バレちゃだめなやつは数値なら12345とかpasswordなど適当なものに変えてもらってOKなので

KS-you commented:
@mo-ri-regen 
コメントありがとうございます。
envの件は了解しました。
__init__.py のuserやpassword等は、.envでの環境変数がうまくいったので修正します。
app.py もcompose.yaml にdebugの設定等がされていたみたいなので、確認して修正します。

KS-you committed:
Delete back/stretch_app directory

stretch_appは不要なディレクトリなので削除

KS-you commented:
@mo-ri-regen 
stretch_app は不要なディレクトリなので削除しました。

KS-you committed:
.env_exampleの作成

KS-you commented:
@mo-ri-regen 
.envファイルのサンプルになります。
確認お願いします。

mo-ri-regen commented on .env_example:
```diff
@@ -0,0 +1,47 @@
+1. flaskコンテナから出てdbコンテナへ入る（3.のコマンドの"flask"を"db"に変えて実施）
+2. mysql -uroot -p でmysqlにログイン（パスワード=password）
+## mysqlにログイン後は、コードの末尾に;を付ける（今回はコメントアウトするのせ省略）
+3. show databases でflasの環境変数に指定したDBがあるか確認
```
「コメントアウトするのせ」→「コメントアウトするので」
ですかね？

35行目
flas→flask

mo-ri-regen commented on .env_example:
```diff
@@ -0,0 +1,47 @@
+
+# Dockerの起動
+1. docker compoes up でflaskとDBを立ち上げた状態で、ブラウザに入り「localhost:5001」でflaskでアプリが立ち上げ、
+   新規登録、ログインができるか確認する。
```
compoes→compose

mo-ri-regen commented on .env_example:
```diff
@@ -0,0 +1,47 @@
+
+# データベースのマイグレーション（移行）方法
+1. docker compoe build を実施
+2. docker compose up -d を実施
```
compoe→compose

mo-ri-regen commented:
@KS-you 
envファイルに全体的に誤字が多いので見直してもらえると🙏
(VS Codeで誤字があったら波線とかになった気がします。もしくは誤字チェックの拡張機能とかあるかも)

話がちょっと変わるのですが
たとえば2行目とか変数以外のところに説明を書いて動くんですかね？
ドキュメント関係はREADMEにまとめといたほうがいい気がします

mo-ri-regen commented on .env_example:
```diff
@@ -0,0 +1,47 @@
+# flaskの設定
+1. back/appディレクトリ内に.envファイルを作成
+2. .envファイル内に以下のように環境変数を作成する。
+
```
コメントなしで文章をenvに書いて動くのでしょうか？
READMEにドキュメント関係はまとめたほうがいい気がします

KS-you commented:
@mo-ri-regen 
確認ありがとうございます。
誤字見直しておきます。
.envファイルについてはコメントアウト（#）しておけば特に問題はありません。
バックエンド側のモジュールで「2.」で説明したような感じでコピーすればプログラムが動くようになっていますが指摘の通りわかりずらいと思ったので、READMEに追記しますね。

KS-you committed:
追跡不要ファイルの追加
static：ディレクトリ（テストcssファイルを格納）
templates:ディレクトリ（テストhtmlファイルを格納）
debug.txt（バックエンド用のメモ）

KS-you commented:
@mo-ri-regen 
.gitignoreファイルに追跡不要のディレクトリとファイルを追加しました。

KS-you committed:
READMEの編集
.env_exampleの修正

KS-you committed:
Update .env_example

誤字の修正

KS-you committed:
Update README.md

記載内容の修正

KS-you committed:
Update README.md

記載内容の修正

KS-you committed:
Update .env_example

記載内容の修正

KS-you commented:
@mo-ri-regen 
flaskの起動方法の修正を行いました確認お願いします。
修正内容：
.env_exampleの誤字、記載内容の修正
READMEの編集（バックエンドで使用する環境変数の追加、flaskに起動方法の案内）

mo-ri-regen commented:
@KS-you 
コンフリクト起きちゃっているのでマージできないです🥺

KS-you committed:
コンクリフトの解消

KS-you committed:
Revert "コンクリフトの解消"

This reverts commit 5e348d23e55a1856ee7ead92f467842a26846655.

KS-you committed:
Delete .gitignore

再度pushし直します。

KS-you committed:
.gitignoreの修正

KS-you committed:
Merge branch 'develop' into you#21
